packages/hierarchies-react/src/presentation-hierarchies-react/UseSelectionHandler.ts

```diff
@@ -0,0 +1,165 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useEffect, useRef } from "react";
+import { isPresentationHierarchyNode, PresentationTreeNode } from "./Types";
+
+/** @internal */
+export enum SelectionModeFlags {
+  Single = 1 << 0,
+  DeselectEnabled = 1 << 1,
+  KeysEnabled = 1 << 2,
+  None = 1 << 3,
+}
+
+/** @beta */
+export enum SelectionMode {
+  /** Only one item selected at a time. */
+  Single = SelectionModeFlags.Single,
+  /** Only one item selected at a time; allows deselecting. */
+  SingleAllowDeselect = SelectionModeFlags.Single | SelectionModeFlags.DeselectEnabled,
+  /** Toggles items. */
+  Multiple = SelectionModeFlags.DeselectEnabled,
+  /** Toggles items; allows the use of Ctrl & Shift Keys. */
+  Extended = SelectionModeFlags.KeysEnabled | SelectionModeFlags.DeselectEnabled,
+  /** Disables selection */
+  None = SelectionModeFlags.None,
+}
+
+/** @internal */
+export type SelectionChangeType = "add" | "replace" | "remove";
+
+interface UseSelectionHandlerProps {
+  rootNodes: Array<PresentationTreeNode> | undefined;
+  selectNode: (nodeIds: Array<string>, changeType: SelectionChangeType) => void;
+  selectionMode: SelectionMode;
+}
+
+interface useSelectionHandlerResult {
+  onNodeClick: (nodeId: string, isSelected: boolean, event: React.MouseEvent<HTMLElement, MouseEvent>) => void;
+  onNodeKeyDown: (nodeId: string, isSelected: boolean, event: React.KeyboardEvent<HTMLElement>) => void;
+}
+
+interface FlatTreeState {
+  flatNodeList: Array<string>;
+  nodeIdToIndexMap: Map<string, number>;
+}
+
+/** @beta */
+export function useSelectionHandler(props: UseSelectionHandlerProps): useSelectionHandlerResult {
+  const { rootNodes, selectionMode, selectNode } = props;
+  const selectionModeRef = useLatest(selectionMode);
```

Comment:
Enum is primitive value so there is no need to wrap it into `useRef`

Reply:
Removed.